src/encoding.rs

```diff
@@ -159,6 +165,53 @@ fn decode_varint_slice(bytes: &[u8]) -> Result<(u64, usize), DecodeError> {
     Err(DecodeError::new("invalid varint"))
 }
 
+/// Decodes a LEB128-encoded variable length integer from the slice, returning the value and the
+/// number of bytes read.
+///
+/// Based loosely on [`ReadVarint64FromArray`][1].
+///
+/// ## Safety
+///
+/// The caller must ensure that `bytes` is non-empty and either `bytes.len() >= 10` or the last
+/// element in bytes is < `0x80`.
+///
+/// [1]: https://github.com/google/protobuf/blob/3.3.x/src/google/protobuf/io/coded_stream.cc#L365-L406
+#[inline]
+#[cfg(target_pointer_width = "64")]
+unsafe fn decode_varint_slice(bytes: &[u8]) -> Result<(u64, usize), DecodeError> {
```

Comment:
(The `step` macro is great, thanks for doing that)

Reply:
> Other than the unchecked accesses, is the only difference here that the results are aggregated into a u64 instead of a pair of u32s?

Yes. I'll remove the 32 bit version.